twilight-model/src/channel/message/kind.rs

```diff
@@ -56,6 +60,65 @@ pub enum MessageType {
     Unknown(u8),
 }
 
+impl MessageType {
+    /// Whether the message can be deleted, not taking permissions into account.
+    /// Some message types can't be deleted, even by server administrators.
+    ///
+    /// Some message types can only be deleted with certain permissions. For
+    /// example, [`AutoModerationAction`][`Self::AutoModerationAction`] can only
+    /// be deleted if the user has the
+    /// [Manage Messages] permission.
+    ///
+    /// To check whether a message can be deleted while taking permissions into
+    /// account, use
+    /// [`deletable_by_permissions`][`Self::deletable_by_permissions`].
+    ///
+    /// [Manage Messages]: Permissions::MANAGE_MESSAGES
+    pub const fn deletable(self) -> bool {
+        matches!(
+            self,
+            Self::Regular
+                | Self::ChannelMessagePinned
+                | Self::UserJoin
+                | Self::GuildBoost
+                | Self::GuildBoostTier1
+                | Self::GuildBoostTier2
+                | Self::GuildBoostTier3
+                | Self::ChannelFollowAdd
+                | Self::ThreadCreated
+                | Self::Reply
+                | Self::ChatInputCommand
+                | Self::GuildInviteReminder
+                | Self::ContextMenuCommand
+                | Self::AutoModerationAction
+        )
+    }
+
+    /// Whether the message can be deleted, taking permissions into account.
+    /// Some message types can't be deleted, even by server administrators.
+    ///
+    /// Some message types can only be deleted with certain permissions. For
+    /// example, [`AutoModerationAction`][`Self::AutoModerationAction`] can only
+    /// be deleted if the user has the [Manage Messages] permission.
+    ///
+    /// To check whether a message can be deleted *without* taking permissions
+    /// into account, use [`deletable`][`Self::deletable`].
+    ///
+    /// [Manage Messages]: Permissions::MANAGE_MESSAGES
+    pub const fn deletable_by_permissions(self, permissions: Permissions) -> bool {
```

Comment:
nit: `by` sounds off to me
```suggestion
    pub const fn deletable_with_permissions(self, permissions: Permissions) -> bool {
```